concert/src/lib.rs

```diff
@@ -0,0 +1,204 @@
+#![no_std]
+
+use concert_io::*;
+use gstd::{msg, prelude::*, ActorId};
+
+const ZERO_ID: ActorId = ActorId::new([0u8; 32]);
+const NFT_COUNT: u128 = 1;
+
+#[derive(Debug, Default)]
+pub struct Concert {
+    pub owner_id: ActorId,
+    pub contract_id: ActorId,
+
+    pub name: u128,
+    pub creator: ActorId,
+    pub number_of_tickets: u128,
+    pub date: u128,
+
+    pub buyers: BTreeSet<ActorId>,
+
+    pub id_counter: u128,
+    // user to token id to metadata
+    pub metadata: BTreeMap<ActorId, BTreeMap<u128, Option<TokenMetadata>>>,
+}
+
+static mut CONTRACT: Option<Concert> = None;
+
+#[no_mangle]
+pub unsafe extern "C" fn init() {
+    let config: InitConcert = msg::load().expect("Unable to decode InitConfig");
+    let concert = Concert {
+        owner_id: config.owner_id,
+        contract_id: config.mtk_contract,
+        ..Concert::default()
```

Comment:
`Default::default()` is more canonic.
```suggestion
        ..Default::default()
```